### PR TITLE
Remove experimental multicluster istioctl cmd

### PR DESCRIPTION
**Pull request by carolynhu**
#29153

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: istioctl/pkg/multicluster/remote_secret.go [hunk 1]
```diff
@@ -1,588 +0,0 @@
...
-	return remoteSecretPrefix + clusterName
-}
-
-func clusterNameFromRemoteSecretName(name string) string {
-	return strings.TrimPrefix(name, remoteSecretPrefix)
-}
-
-// NewCreateRemoteSecretCommand creates a new command for joining two contexts
-// together in a multi-cluster mesh.
-func NewCreateRemoteSecretCommand() *cobra.Command {
```
Comment (howardjohn) [inline istioctl/pkg/multicluster/remote_secret.go:86]: This is still used and essential

File: releasenotes/notes/remove-addons-mixer-istioctl.yaml [hunk 2]
```diff
@@ -1,11 +1,9 @@
 apiVersion: release-notes/v2
 kind: feature
-area: installation
+area: istioctl
```
Comment (howardjohn) [inline releasenotes/notes/remove-addons-mixer-istioctl.yaml:3]: you are replacing the wrong file

**Review by howardjohn — COMMENTED** [review]
cc @stevenctl @nmittler can you confirm the other MC commands are not used?

File: releasenotes/notes/remove-addons-mixer-istioctl.yaml (hunk 2 above)
Comment (carolynhu) [reply releasenotes/notes/remove-addons-mixer-istioctl.yaml:3]: oops...

File: istioctl/pkg/multicluster/remote_secret.go (hunk 1 above)
Comment (carolynhu) [reply istioctl/pkg/multicluster/remote_secret.go:86]: probably I should first only remove "apply" "describe" and "generate" first for safety

**Comment by carolynhu** [comment]
i got `No space left on device` when i run `make gen`,  this caused by macos Catalina upgrade, I need to figure that out to rerun `make gen`...

File: istioctl/pkg/multicluster/cluster.go [hunk 3]
```diff
@@ -61,153 +36,6 @@ func clusterUID(client kubernetes.Interface) (types.UID, error) {
 	return kubeSystem.UID, nil
 }
```
Comment (carolynhu) [inline istioctl/pkg/multicluster/cluster.go:38]: keep those in the cluster.go file that are still used/referenced by other places in the codebase

**Comment by nmittler** [comment]
@ayj @stevenctl while there are no tests today using these commands, I suspect there may be some usefulness in them. Just taking a scan through, the describe command looks like it may provide some additional debugging information for users.

Perhaps some of this could be made more general within istioctl and not multi-cluster specific?

**Comment by stevenctl** [comment]
I think the describe command could be useful.. not sure about the `apply` one though.. seems like it hides a lot of details and it's unlikely a user would update all clusters at once like this.

**Comment by carolynhu** [comment]
@nmittler @stevenctl if `describe` command is useful, should i revert it back?

**Comment by howardjohn** [comment]
My reading of describe is it requires some file of shape
```
type Mesh struct {
meshID                string
clustersByContext     map[string]*Cluster // by Context
clustersByClusterName map[string]*Cluster
}
```

which we don't really have anymore?

**Review by nmittler — APPROVED** [review]
I've fine removing for now, since we're not using it. If any of this is useful, we can always resurrect later.

**Pull request merged**